sonar-plugin-api/src/main/java/org/sonar/api/batch/scm/ScmProvider.java

```diff
@@ -64,4 +64,19 @@ public BlameCommand blameCommand() {
   public Set<Path> branchChangedFiles(String targetBranchName, Path rootBaseDir) {
     return null;
   }
+
+  /**
+  * The relative path to the project from SCM root, typically ".".
+  */
+  public String projectBasedir() {
+    return ".";
```

Comment:
I don't like this default value. "." doesn't have a meaning in itself. If we are talking about returning the relative path of sqProjectBaseDir comparing to scmRoot, we could return Path.get("").

Reply:
Another option is to throw UnsupportedOperationException and catch it in the calling code, to use the appropriate default value.